[AIRFLOW-5533] Fixed failing CRON build

potiuk opened the pull request:
This change optimises further image building and removes unnecessary
verbosity in building the images for CI builds.

After this change is merged, only the necessary images are built for
each type of check:
* Tests -> only CI
* Static checks (with/without pylint) -> Only CI_SLIM
* Docs -> only CI_SLIM
* Licence checks -> Only CHECKLICENCE

Previously the right images only were built in ci_before_install.sh
but then in case of static checks, the pre-commit build image step
also rebuilt CHECKLICENCE and CI images - which was not necessary
and very long in case of CRON job - this caused the CRON job to
fail at 10m of inactivity.

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

potiuk committed:
[AIRFLOW-5534] Less verobosity and removal of context container

potiuk committed:
[AIRFLOW-5533] Fixed failing CRON build

This change optimises further image building and removes unnecessary
verbosity in building the images for CI builds.

After this change is merged, only the necessary images are built for
each type of check:
* Tests -> only CI
* Static checks (with/without pylint) -> Only CI_SLIM
* Docs -> only CI_SLIM
* Licence checks -> Only CHECKLICENCE

Previously the right images only were built in ci_before_install.sh
but then in case of static checks, the pre-commit build image step
also rebuilt CHECKLICENCE and CI images - which was not necessary
and very long in case of CRON job - this caused the CRON job to
fail at 10m of inactivity.

potiuk committed:
[AIRFLOW-5535] Fix name of VERBOSE parameter

mik-laj commented on scripts/ci/_utils.sh:
```diff
@@ -885,3 +901,32 @@ function rebuild_all_images_if_needed_and_confirmed() {
+        rebuild_ci_image_if_needed
+    elif [[ ${TRAVIS_JOB_NAME} == "Check lic"* ]]; then
+        rebuild_checklicence_image_if_needed
+    else
```
Can you specify the cases when this image is build and the other case returns an error? This may allow easier script debugging.

potiuk replied:
Fixed :)

potiuk committed:
[AIRFLOW-5536] Better handling of temporary output files

potiuk committed:
[AIRFLOW-5537] Yamllint is not needed as dependency on host

It used to be needed for pre-commits but is not needed any more
as it is automatically installed as dependency in the virtualenv
created by pre-commit
